portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -36,9 +36,25 @@
   @Override
   public void init() {
       messages = new ArrayList<String>();
-      messages.add("Hello, this is Martin!");
-      messages.add("How is your day?");
-      messages.add("Did you learn something new today?");
+  }
+
+  /**
+    * Receive an input from frontend.
+    */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+      //Get the input from the form.
+      String comment = request.getParameter("textInput");
+
+      //Add input to messages array.
```

Comment:
nit: While this looks pretty clean in this short method, this is actually a great example of overusing comments. These basic one-liner calls are themselves pretty self-explanatory (e.g. `messages.add(comment);` is pretty clear).

In general, prefer using comments to describe logical blocks of multiple lines, and explaining why you're doing something, if it isn't obvious.